pkg/controller/performanceprofile/performanceprofile_controller.go

```diff
@@ -103,54 +210,350 @@ func (r *ReconcilePerformanceProfile) Reconcile(request reconcile.Request) (reco
+func (r *ReconcilePerformanceProfile) applyComponents(profile *performancev1alpha1.PerformanceProfile) (*reconcile.Result, error) {
+	// get mutated machine config pool
+	mcp := machineconfigpool.New(profile)
+	if err := controllerutil.SetControllerReference(profile, mcp, r.scheme); err != nil {
+		return nil, err
 	}
-	return &corev1.Pod{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      cr.Name + "-pod",
-			Namespace: cr.Namespace,
-			Labels:    labels,
-		},
-		Spec: corev1.PodSpec{
-			Containers: []corev1.Container{
-				{
-					Name:    "busybox",
-					Image:   "busybox",
-					Command: []string{"sleep", "3600"},
-				},
-			},
-		},
+	mcpMutated, created, err := r.getMutatedMachineConfigPool(mcp)
+	if err != nil {
+		return nil, err
+	}
+
+	// get mutated machine config
+	mc, err := machineconfig.New(r.assetsDir, profile)
+	if err != nil {
+		return nil, err
+	}
+	if err := controllerutil.SetControllerReference(profile, mc, r.scheme); err != nil {
+		return nil, err
+	}
+	mcMutated, err := r.getMutatedMachineConfig(mc)
+	if err != nil {
+		return nil, err
+	}
+
+	// get mutated feature gate
+	fg := featuregate.NewLatencySensitive()
+	// TOOD: uncomment once https://bugzilla.redhat.com/show_bug.cgi?id=1788061 fixed
+	// if err := controllerutil.SetControllerReference(profile, fg, r.scheme); err != nil {
+	// 	return err
+	// }
+	fgMutated, err := r.getMutatedFeatureGate(fg)
+	if err != nil {
+		return nil, err
+	}
+
+	// get mutated kubelet config
+	kc, err := kubeletconfig.New(profile)
+	if err != nil {
+		return nil, err
+	}
+	if err := controllerutil.SetControllerReference(profile, kc, r.scheme); err != nil {
+		return nil, err
+	}
+	kcMutated, err := r.getMutatedKubeletConfig(kc)
+	if err != nil {
+		return nil, err
+	}
+
+	// get mutated network latency tuned
+	networkLatencyTuned, err := tuned.NewNetworkLatency(r.assetsDir)
+	if err != nil {
+		return nil, err
+	}
+	if err := controllerutil.SetControllerReference(profile, networkLatencyTuned, r.scheme); err != nil {
+		return nil, err
+	}
+	networkLatencyTunedMutated, err := r.getMutatedTuned(networkLatencyTuned)
+	if err != nil {
+		return nil, err
+	}
+
+	// get mutated real time kernel tuned
+	realTimeKernelTuned, err := tuned.NewWorkerRealTimeKernel(r.assetsDir, profile)
+	if err != nil {
+		return nil, err
+	}
+	if err := controllerutil.SetControllerReference(profile, realTimeKernelTuned, r.scheme); err != nil {
+		return nil, err
+	}
+	realTimeKernelTunedMutated, err := r.getMutatedTuned(realTimeKernelTuned)
+	if err != nil {
+		return nil, err
+	}
+
+	// does not update any resources, if it no changes to relevant objects and just continue to the status update
+	if mcpMutated == nil &&
+		mcMutated == nil &&
+		kcMutated == nil &&
+		fgMutated == nil &&
+		networkLatencyTunedMutated == nil &&
+		realTimeKernelTunedMutated == nil {
+		// if nothing needed to be updated and machine config pool paused, unpause it
+		mcpUpdate, err := r.getMachineConfigPool(mcp.Name)
```

Comment:
the mcp object has references to objects in the status section.

the ```mcp.status.configuration.source``` list. Can we look at that to make sure the mcp picked up all our configs before unpausing it?

Reply:
mm feature gate and kubelet config machine configs rendered always with the same name, so I unsure how can we check it, see - https://github.com/openshift/machine-config-operator/blob/04cd2198cae247fabcd3154669618d74f124f27f/pkg/controller/kubelet-config/helpers.go#L60

Reply:
so, we'd only be able to detect the new MC was applied?

here's an issue  #35 lets follow up on this discussion and see if there's a way to detect when unpause is safe. or at least detect the mc we posted has been picked up before unpausing.